### PR TITLE
Insufficient storage error

### LINKED ISSUE #1767: Error message for Insufficient Storage error
### Actual behaviour

I have enabled automatic image upload on my Android owncloud app. Suddenly I got `Unknown error` and the image could not be uploaded. When I looked at the server error log, I read the actual error message: `507 Insufficient Storage`.
### Expected behaviour

Why is not the server error used, which explains exactly what is going on. `Unknown error` is not very descriptive.
### Steps to reproduce
1. Enable automatic image upload on your Android app
2. Set a too low quota on the server
3. Take a picture with your phone and wait for the automatic image upload

I don't know if this problem is specific to the photo upload. Maybe it concerns all upload when there is no space available.
### Environment data

Android version: Android 5

Device model: Huawei Honor 7

Stock or customized system: Stock

ownCloud app version:2.0.1

ownCloud server version: 9.0.3
#### ownCloud log (data/owncloud.log)

```
Exception: {"Message":"HTTP\/1.1 507 Insufficient Storage","Exception":"Sabre\\DAV\\Exception\\InsufficientStorage","Code":0,"Trace":"#0 [internal function]: OCA\\DAV\\Connector\\Sabre\\QuotaPlugin->checkQuota('photos\/Honor 7\/...', Resource id #31, Object(OCA\\DAV\\Connector\\Sabre\\Directory), false)\n#1 \/var\/www\/owncloud\/3rdparty\/sabre\/event\/lib\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\n#2 \/var\/www\/owncloud\/3rdparty\/sabre\/dav\/lib\/DAV\/Server.php(1034): Sabre\\Event\\EventEmitter->emit('beforeCreateFil...', Array)\n#3 \/var\/www\/owncloud\/3rdparty\/sabre\/dav\/lib\/DAV\/CorePlugin.php(523): Sabre\\DAV\\Server->createFile('photos\/Honor 7\/...', Resource id #31, NULL)\n#4 [internal function]: Sabre\\DAV\\CorePlugin->httpPut(Object(Sabre\\HTTP\\Request), Object(Sabre\\HTTP\\Response))\n#5 \/var\/www\/owncloud\/3rdparty\/sabre\/event\/lib\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\n#6 \/var\/www\/owncloud\/3rdparty\/sabre\/dav\/lib\/DAV\/Server.php(459): Sabre\\Event\\EventEmitter->emit('method:PUT', Array)\n#7 \/var\/www\/owncloud\/3rdparty\/sabre\/dav\/lib\/DAV\/Server.php(248): Sabre\\DAV\\Server->invokeMethod(Object(Sabre\\HTTP\\Request), Object(Sabre\\HTTP\\Response))\n#8 \/var\/www\/owncloud\/apps\/dav\/appinfo\/v1\/webdav.php(55): Sabre\\DAV\\Server->exec()\n#9 \/var\/www\/owncloud\/remote.php(138): require_once('\/var\/www\/ownclo...')\n#10 {main}","File":"\/var\/www\/owncloud\/apps\/dav\/lib\/connector\/sabre\/quotaplugin.php","Line":108,"User":"filtered_user_name"}
```

### PR DESCRIPTION
**Pull request by davigonz**
Fix #1767

~~Requires https://github.com/owncloud/android-library/pull/147~~

### CODE REVIEW

**Comment by davigonz** [comment]
@davivel , @jesmrec this bug is fixed, pending to review and pass QA

File: res/values/strings.xml [hunk 1]
```diff
@@ -465,6 +465,7 @@
     <string name="confirmation_remove_files_alert">Do you really want to remove the selected items?</string>
     <string name="confirmation_remove_folders_alert">Do you really want to remove the selected items and their contents?</string>
     <string name="maintenance_mode">Server in maintenance mode</string>
+    <string name="failed_upload_quota_exceeded_text">Not enough free space</string>
```
Comment (davivel) [inline res/values/strings.xml]: We should specify in some way that it's a lack of space ** in the server **. I think it's important. Maybe

"Not enough free space in server", or "Not enough space in server" if the first is too long. 

cc @jesmrec for second opinion.

File: src/com/owncloud/android/ui/errorhandling/ErrorMessageAdapter.java [hunk 2]
```diff
@@ -119,33 +119,35 @@ private static String getSpecificMessageForResultAndOperation(
 
             if (result.isSuccess()) {
                 message = String.format(
-                    res.getString(R.string.uploader_upload_succeeded_content_single),
-                    ((UploadFileOperation) operation).getFileName());
+                        res.getString(R.string.uploader_upload_succeeded_content_single),
```
Comment (davivel) [inline src/com/owncloud/android/ui/errorhandling/ErrorMessageAdapter.java]: Please, remind me to compare our autoformat settings.

**Comment by davivel** [comment]
Just an easy one.

File: res/values/strings.xml (hunk 1 above)
Comment (davigonz) [inline res/values/strings.xml]: I've copied the message from iOS solution few days ago but I've just checked the code of a similar issue on iOS and has changed, now is: "Not enough free space in your account". This is a proper message. What do you think?

Comment (davivel) [inline res/values/strings.xml]: Seems fine for the notification, but seems too long for the uploads view. iOS prints all that text in the uploads view?

Comment (jesmrec) [inline res/values/strings.xml]: agreed with the comments. it's important to clarify that the displayed problem is not caused by the device storage. "Account" or "Server" are both correct options IMHO.

Comment (davivel) [inline res/values/strings.xml]: @davigonz, check if that message fits in a single line in the uploads view. I have a bad feeling about it. 

iOS displays the message in a single line, we show more information in the same line. 

Could try to equalize the items in both platforms, but I wouldn't like this grows too much in UI changes.

**Comment by jesmrec** [comment]
Approved

**Pull request merged**